doc/arch/adr-003-data-retention.md

```diff
@@ -0,0 +1,287 @@
+datatypes. This will allow us to remove the reference field of
+`delivery_attempt` - which seems to be created as a psuedo uuid value
+combining a database id and a timestamp. The reason for changing email to be a
+uuid is to have this as a value that is globally unique and can be used as a
+further reference without revealing system information.
+
+#### SubscriptionContent
+
+SubscriptionContent is currently the glue that holds together the association
+between a ContentChange, a Subscription, a resulting Email and a
+DigestRunSubscriber. Currently when an email is deleted there is a cascade
+to set the value to SubscriptionContent to null. This is incorrect as the value
+of null here indicates an email is scheduled to be generated. Instead this
+foreign key should be set to have an on delete cascade association with Email
+so that when an email is removed from the system the proceeding
+SubscriptionContent values are also deleted.
+
+This will mean that over time the data in the SubscriptionContent will be
+culled routinely as part of the Email deletion process.
+
+#### DigestRun and DigestRunSubscriber
+
+Currently we store each DigestRun and an association to each subscriber. If we
+want to know how many subscribers were due to be emailed about a digest we can
+count the number of DigestRunSubscibers associated.
+
+As the number of subscribers related with a digest is something that is known
+at the point of digest creation we can store this as a count on
+DigestRun. This can alleviate the need for storing each DigestRunSubscriber
+(which can be removed at approximately the same interval as Emails).
+
+As DigestRuns are low on data usage they can effectively be stored indefinitely
+or at least until we determine an approach to archiving them.
+
+#### ContentChange and SubscriptionList
+
+Both ContentChange and SubscriptionList should not require changes to them in
+order to allow us to build up a full analytics picture. However they would
+require an intention not to delete either of them.
+
+As a SubscriberList can be updated the only way we can determine the historical
+number of ContentChanges with accuracty is to have the ContentChanges
+available (to perform the look up between them).
+
+It is likely that over time ContentChanges may reach an infeasible number and
+it will then need to be looked at as to what value storing them maintains, but
+hopefully by that point (expected to be a couple of years away) we'll have a
+much firmer understanding of the analytics usage and needs.
+
+SubscriptionLists don't currently have the ability to be deleted by apps, and
+it's likely if they gained that ability that would need to be some form of
+archiving so it could still be communicated to users that the list they had a
+subscription to is no longer available.
+
+#### Subscriber and Subscription
+
+Currently when a subscription is ended or we discover an email address is not
```

Comment:
We don't actually delete the subscriber currently I don't think, we just nullify their address. We're also currently only 'soft deleting' the subscriptions which we intended to remove but it sounds like we probably want to keep that. So... I don't think we need some of the changes proposed here.

Reply:
Yup you're right. I'll adjust the text.